Several improvements and updates for notification and log report

fepitre opened the pull request.

marmarek commented on github-action.py:
```diff
@@ -755,27 +1250,67 @@ def upload(self):
                     status="failed",
                     stage="upload",
                     log_file=autobuild_exc.log_file,
+                    additional_info=extra,
```
Pattern of those two calls, with very similar if not identical arguments, being together repeats over and over. Maybe `update_result` could optionally call `notify_build_status`, or the other way around? Based on some extra argument like `notify=True/False`.

fepitre replied:
Yes I can do that. The idea was to first have a review and then improve.

marmarek commented on github-action.py:
```diff
@@ -854,20 +1392,26 @@ def run_stages(self, stages):
+                stage=result.stage or "build",
+                reason="Timeout",
+                additional_info="Timeout",
+            )
```
This function now is not just 1-liner repeated in each class, but 10-liner. And it seems the only difference is the argument to `self.get_result()`. Please deduplicate.

marmarek commented on github-action.py:
```diff
@@ -692,41 +1121,107 @@ def build(self):
+                self.notify_build_status(
+                    status="failed",
+                    stage=stage,
+                    additional_info=extra,
```
Those two cases are very similar, maybe they can be merged?
In fact, when `QubesBuilderError` could happen at all? You already modified `make_with_log_local`/`make_with_log_qrexec` to catch broad `Exception`, not just `PluginError`.

marmarek commented on github-command.py:
```diff
@@ -277,9 +289,445 @@ def main():
 
 
+#
+# action subcommand
```
Why merging two (three?) scripts into one, with basically no shared parts? This forced you to do local imports in `_run_action`, which isn't really nice. Separate scripts required less hacks...

fepitre replied:
I found that it was "convenient" to maintain only one place for CLI as now we have separate libraries directly.

fepitre replied:
I've simplified everything here.

marmarek reviewed (changes requested):
Generally I like the direction where it's going, but the current shape added quite a bit of code duplication. Specific comments below.

marmarek commented on github-command.py:
```diff
@@ -277,9 +243,473 @@ def main():
+#
+
+
+def _run_notify(args):
```
Is this (and the relevant parser) even used anywhere (outside of tests)? action.py creates `NotifyIssueCli` instance directly.

fepitre replied:
Nope useless now I missed it in the today refactor, let me fix that

fepitre replied:
I spoke too fast, it's only for creating the parser related to the commands, like others. You may want to use notify like the previous CLI tool.

fepitre committed:
Refactor all the CLIs and split tests that needs fresh fixtures

- pytest: make the tests verbose and add logs to artifacts
- ci: stream logging and add helper scripts for tests
